src/Components/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { ErrorBlockProps, ErrorBoundaryProps, ErrorBoundaryState } from './models'
+import './styles.scss'
+
+const ErrorBlock: React.FC<ErrorBlockProps> = ({error}) => {
+    return (
+        <div className="error-boundary">
+            <div className="error-boundary__body">
+                <div className="error-boundary__title">
+                    <h1>Something went wrong</h1>
+                    <hr />
+                    <div className="error-boundary__message">
+                        <>{error?.message}</>
+                        <>{error?.stack}</>
+                    </div>
+                </div>
+            </div>
+        </div>
+    )
+}
+
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
```

Comment:
Not an error, but you can use `React.PropsWithChildren<P>`:
```js
type ComponentWithChildProps = React.PropsWithChildren<{}>;
```

Reply:
Got it, thank you